### ISSUE TITLE
Out-of-the-box gulp build fails

### ISSUE THREAD

**Issue by james-nash** [pattern-lab/edition-node-gulp#54]
I am using Pattern Lab Node `v2.0.0` on `Mac`, using the `gulp` configuration.
##### Expected Behavior

Running `gulp` after the initial setup should complete successfully
##### Actual Behavior

The `build` task errors with the following message:

```
...
[15:51:21] Finished 'pl-assets' after 87 ms
[15:51:21] Starting 'build'...
[15:51:21] 'build' errored after 181 ms
[15:51:21] TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.extname (path.js:1433:5)
    at /Users/ja310147/tmp/edition-node-gulp/node_modules/patternlab-node/core/lib/media_hunter.js:13:16
    at diveSync (/Users/ja310147/tmp/edition-node-gulp/node_modules/diveSync/diveSync.js:62:5)
    at findMediaQueries (/Users/ja310147/tmp/edition-node-gulp/node_modules/patternlab-node/core/lib/media_hunter.js:12:5)
    at Object.find_media_queries (/Users/ja310147/tmp/edition-node-gulp/node_modules/patternlab-node/core/lib/media_hunter.js:34:7)
    at buildFrontEnd (/Users/ja310147/tmp/edition-node-gulp/node_modules/patternlab-node/core/lib/ui_builder.js:420:16)
    at Object.build (/Users/ja310147/tmp/edition-node-gulp/node_modules/patternlab-node/core/lib/patternlab.js:366:7)
    at build (/Users/ja310147/tmp/edition-node-gulp/gulpfile.js:79:14)
    at bound (domain.js:280:14)
[15:51:21] 'patternlab:build' errored after 271 ms
```
##### Steps to Reproduce
1. Cloned the latest `edition-node-gulp` repo using `git clone git@github.com:pattern-lab/edition-node-gulp.git`
2. Ran `npm install` to fetch all the dependencies
3. Ran `gulp patternlab:build`
##### Notes

I haven't had time to investigate thoroughly, but judging by the messages, this error occurs in `patternlab-node/core/lib/media_hunter.js` so this may be a bug within `patternlab-node` itself, rather than this edition.

FWIW, it looks like it's coming from the code using to `diveSync` that searches for CSS files. I guess the callback function is being called with the `file` argument not being set to a string. Dunno if that's any help.

**Comment by bmuenzenmeyer** [comment]
Can you provide the contents of your `patternlab-config.json` file? Do you have `css` files inside whatever mapped to `.source/css/` ?

**Comment by james-nash** [comment]
At the moment, it's just the vanilla `patternlab-config.json` that came with `edition-node-gulp`.

I just cloned it directly with the intent of trying out Patternlab 2.0. I haven't attempted to incorporate my own project (i.e. mustache files, CSS, etc.) yet at all. It's just plain out of the box at the moment.

Do I need to install a starter kit before it becomes usable perhaps? Alternatively, could it be something about my environment?

Anyway, here's the contents of the config file, just in case:

```
{
  "paths" : {
    "source" : {
      "root": "./source/",
      "patterns" : "./source/_patterns/",
      "data" : "./source/_data/",
      "meta": "./source/_meta/",
      "annotations" : "./source/_annotations/",
      "styleguide" : "./node_modules/styleguidekit-assets-default/dist/",
      "patternlabFiles" : "./node_modules/styleguidekit-mustache-default/views/",
      "js" : "./source/js",
      "images" : "./source/images",
      "fonts" : "./source/fonts",
      "css" : "./source/css/"
    },
    "public" : {
      "root" : "./public/",
      "patterns" : "./public/patterns/",
      "data" : "./public/styleguide/data/",
      "annotations" : "./public/annotations/",
      "styleguide" : "./public/styleguide/",
      "js" : "./public/js",
      "images" : "./public/images",
      "fonts" : "./public/fonts",
      "css" : "./public/css"
    }
  },
  "styleGuideExcludes": [
  ],
  "defaultPattern": "all",
  "defaultShowPatternInfo": false,
  "cleanPublic" : true,
  "patternExtension": "mustache",
  "ignored-extensions" : ["scss", "DS_Store", "less"],
  "ignored-directories" : ["scss"],
  "debug": false,
  "ishControlsHide": {
    "s": false,
    "m": false,
    "l": false,
    "full": false,
    "random": false,
    "disco": false,
    "hay": true,
    "mqs": false,
    "find": false,
    "views-all": false,
    "views-annotations": false,
    "views-code": false,
    "views-new": false,
    "tools-all": false,
    "tools-docs": false
  },
  "ishMinimum": "240",
  "ishMaximum": "2600",
  "patternStateCascade": ["inprogress", "inreview", "complete"],
  "patternStates": {
  },
  "patternExportPatternPartials": [],
  "patternExportDirectory": "./pattern_exports/",
  "cacheBust": true,
  "starterkitSubDir": "dist"
}
```

**Comment by bmuenzenmeyer** [comment]
Which version of Edition Node Gulp are you on? 1.2.3 fixed an issue with `gulp patternlab:build` but it doesn't seem like the exact same error.

**Comment by james-nash** [comment]
According to the `package.json` I'm on 1.2.3. I just cloned the master branch direct from Github instead of downloading a zipped release. Dunno if that would make any difference.

FWIW, my node version is 6.2.1.

I've got `gulp-cli` installed and the versions match up with the gulp upgrade instructions referenced from the README:

```
$ gulp -v
[10:49:55] CLI version 1.2.1
[10:49:55] Local version 4.0.0-alpha.2
```

I was attempting to follow the steps in this edition's README. Perhaps I missed something or made a mistake. I've just tried repeating them from scratch and I get the same result. Specifically, here's all that I've done:
1. Cloned the latest edition-node-gulp repo using `git clone git@github.com:pattern-lab/edition-node-gulp.git`
2. Ran `npm install` to fetch all the dependencies
3. Ran `gulp patternlab:build`

I just tried repeating the above steps in an empty directory and still get the same build error.

**Comment by james-nash** [comment]
Ok, I did a little more digging and I think I found a bug in `patternlab-node` that causes this. In [`media_hunter.js`](https://github.com/pattern-lab/patternlab-node/blob/master/core/lib/media_hunter.js) on line 16, you have a regex looking for media queries in the CSS. However, if there are no matches, it returns `null`. This never gets checked for, so the beginning of the for loop on the next line fails when you try to do `matches.length`.

In other words, it looks like this will always fail when it encounters CSS files that do not contain any media queries.

If I wrap the loop in an `if` statement to check for non-null, my patternlab build completes successfully!

``` js
...
if( matches ){
  for (var i = 0; i < matches.length; i++) {
    var breakpoint = matches[i].substring(matches[i].indexOf(':') + 1).trimLeft();
    if (patternlab.mediaQueries.indexOf(breakpoint) === -1) {
      patternlab.mediaQueries.push(breakpoint);
    }
  }
}
...
```

If you agree that this is a genuine bug that needs fixing, then I'm happy to prepare a proper PR with the fix.
